### DISCUSSION TITLE
How to use the 'Plugins  options' in wails3

### DISCUSSION THREAD

**Discussion by pandason125** [wailsapp/wails#3792] [General]
The documents had inintroduced Plugins  demo ，but application.Options didn‘t use it  like :
`		Plugins: map[string]application.Plugin{
			"single_instance": single_instance.NewPlugin(&single_instance.Config{
				// When true, the original app will be activated when a second instance is launched
				ActivateAppOnSubsequentLaunch: true,
			}),
		},`
the 'Plugins' is promt unknown field Plugins in struct literal of type

**Comment by leaanthony** [comment]
Plugins is deprecated in favour of services.

**Comment by konster** [comment]
then wails3 how to set single instance?wails3 SingleInstanceLock option was deprecated

**Reply by pandason125** [reply]
you can see the example folder ，code segment like this：
`		application.NewService(single_instance.NewPlugin(&single_instance.Config{
				LockFileName:                  "pid.lock",
				LockFilePath:                  path.Join(userdir.GetConfigHome(), consts.PROJECT_NAME),
				ActivateAppOnSubsequentLaunch: true,
			})),`

**Reply by leaanthony** [reply]
The plugins will be ported to services. Don't try and use the plugins.
